src/pages/Main/Components/ProjectCards.js

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect, useRef } from 'react';
+import ProjectCard from './ProjectCard';
+import Button from '../../../components/Button/Button';
+import { useFilterContext } from '../Main';
+import styled from 'styled-components/macro';
+
+const SORT = [
+  {
+    0: {
+      url: 'Latest',
+      kor: '최신순',
+    },
+  },
+  {
+    1: {
+      url: 'People',
+      kor: '최다 후원순',
+    },
+  },
+  {
+    2: {
+      url: 'Amount',
+      kor: '최다 금액순',
+    },
+  },
+  {
+    3: {
+      url: 'Old',
+      kor: '마감 임박순',
+    },
+  },
+];
+
+export default function ProjectCards() {
+  const { filterOption } = useFilterContext();
+  const [mainData, setMainData] = useState([]);
+  const [isOpened, setIsOpened] = useState(false);
+  const [clickedSortOption, setClickedSortOption] = useState(SORT[0][0]);
+  const nextCount = useRef(6);
+
+  useEffect(() => {
+    fetch(`/data/Main/MainData${clickedSortOption.url}.json`)
+      .then(res => res.json())
+      .then(({ projects }) =>
+        setMainData(projects.slice(0, nextCount.current))
+      );
+  }, [clickedSortOption]);
+
+  const getMoreData = () => {
+    fetch(`/data/Main/MainData${clickedSortOption.url}.json`)
+      .then(res => res.json())
+      .then(({ projects }) => {
+        if (mainData.length === projects.length) {
+          alert('더이상 불러올 데이터가 없습니다');
+          return;
+        }
+        const moreData = projects.slice(
+          nextCount.current,
+          nextCount.current + 6
+        );
+        nextCount.current += 6;
+        setMainData([...mainData, ...moreData]);
```

Comment:
```suggestion
        setMainData(prev => [...prev, ...moreData]);
```
외부의 값에 의존하기 보다 함수 업데이트를 통해 이전 값을 활용하도록 하면 더 좋을거 같네요!

Reply:
오오...감사합니다!